scrapy/utils/conf.py

```diff
@@ -13,7 +15,7 @@ def build_component_list(base, custom):
     """
     if isinstance(custom, (list, tuple)):
         return custom
-    compdict = base.copy()
+    compdict = BaseSettings(base, priority = 'default')
```

Comment:
A note about coding style in this line and in general: though we're not enforcing it we try to comply with [pep8](https://www.python.org/dev/peps/pep-0008/) for new added code to scrapy as closely as we can (only rule we agreed to change it's the 80 chars max line width rule, we settled on a more flexible 100 chars). This pull request follows pep8 mostly, but I'd try to stick to the rule of "There shouldn't be spaces around default arguments nor immediately after parenthesis, brackets or braces".